modules/Rock.cmake

```diff
@@ -49,6 +49,30 @@ function(rock_add_compiler_flag_if_it_exists FLAG)
     endif()
 endfunction()
 
+macro(rock_add_target_properties _target _name _properties)
+    set(_new_properties ${_properties} ${ARGN})
+    get_target_property(_old_properties ${_target} ${_name})
+    if(NOT _old_properties)
+        set(_old_properties)
+    else()
+        set(_old_properties "${_old_properties} ")
+    endif()
+    string (REPLACE ";" " " _new_properties "${_old_properties}${_new_properties}")
```

Comment:
I'm fairly sure you both don't need that, and that it is actually harmful. Lists in CMake are semicolon-separated, which means that you are actually actively breaking the separation here, which would break if some elements contain a space.

Probably better (untested !):

```
get_target_property(_all_properties ${_target} ${_name})
list(APPEND _all_properties ${ARGN})
set_target_properties(${_target} PROPERTIES ${_name} ${_all_properties})
```

and IMO it would be better to remove the _properties argument altogether (since you use ARGN). It makes the zero-argument a corner-case, which I don't see really why it should be